### ISSUE TITLE
Add FutureContinuation helper functions to the paper

### ISSUE THREAD

**Issue by dhollman** [executors/futures#48]
At the second face-to-face, we decided that `then` will always take a `FutureContinuation`, but that the paper will provide helper functions for these. I'm working on a pull request that will do this now.

**Comment by dhollman** [comment]
I've added wording for these to the paper.  Any issues?

**Comment by brycelelbach** [comment]
From the NVIDIA F2F meeting:

Poll: .then and then_execute take FutureContinuation (conceptually a pair of callables, currently defined as what is in executors), we do error handling/propagation as above (the same thing Executors do), and we don't do unwrapping, we provide some helper functions (then_error, then_value, and then_value_error which don't unwrap) in version 1 (Rapperswil) of the paper.
SF F N A SA
4 3 0 0 0

**Comment by brycelelbach** [comment]
Action Items:
* [ ] Implement wording for this

Sounds like @dhollman has done the necessary work here already in a PR, so I will just need to review that.

**Comment by dhollman** [comment]
@brycelelbach Also, see my comments on #49 concerning the implementability/specify-ability of `on_expected` without an explicit template parameter for the type.

**Comment by griwes** [comment]
Yes, it'll literally pass the value to the callback - but will also make sure that the callback will never be called when the continuation is called in the exceptional case and the callback would match that.

**Comment by dhollman** [comment]
@griwes But not using `on_value` with a lambda also ensures that. I don't see a reason to add unnecessary complexity to v1 of the paper just for syntactic sugar.

**Comment by dhollman** [comment]
@griwes I guess I could see needing this if you wanted to use a `FutureContinuation` from a library but didn't want to use that library's error handling. Is that enough to justify this as a V1 feature, given that you can accomplish the same thing by creating a trivial pass-through lambda?

**Comment by griwes** [comment]
It doesn't ensure that if the callback has a variadic, templated call operator.

**Comment by griwes** [comment]
A pass-through lambda is IMO verbose enough to warrant this in V1.

**Comment by dhollman** [comment]
@griwes I don't have anything against this, but we should check with others pushing for minimalism in v1 before adding this. Is @brycelelbach okay with this, for instance? I'm fine with it, as long as it doesn't end up replacing some other feature that gets removed in the name of minimalism.

**Comment by dhollman** [comment]
(The above comment doesn't apply to `on_expected`, by the way, since I still don't see a way to implement this without an explicit template parameter, which is too ugly for V1 IMHO)

**Comment by dhollman** [comment]
We can close this issue now, since the corresponding PR has been merged.
